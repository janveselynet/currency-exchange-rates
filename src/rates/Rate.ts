export interface Rate {
  country: string;
  currency: string;
  currencyCode: string;
  rate: number;
}
