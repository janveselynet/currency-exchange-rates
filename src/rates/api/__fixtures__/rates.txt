25 Nov 2022 #228
Country|Currency|Amount|Code|Rate
Australia|dollar|1|AUD|15.831
Brazil|real|1|BRL|4.395
Bulgaria|lev|1|BGN|12.473
Canada|dollar|1|CAD|17.559
China|renminbi|1|CNY|3.278
Croatia|kuna|1|HRK|3.235
Hungary|forint|100|HUF|5.923
