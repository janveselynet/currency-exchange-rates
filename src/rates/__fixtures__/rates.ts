import { Rate } from '../Rate';

export const rates: Array<Rate> = [
  {
    country: 'Australia',
    currency: 'dollar',
    currencyCode: 'AUD',
    rate: 15.831,
  },
  {
    country: 'Brazil',
    currency: 'real',
    currencyCode: 'BRL',
    rate: 4.395,
  },
  {
    country: 'Bulgaria',
    currency: 'lev',
    currencyCode: 'BGN',
    rate: 12.473,
  },
  {
    country: 'Canada',
    currency: 'dollar',
    currencyCode: 'CAD',
    rate: 17.559,
  },
  {
    country: 'China',
    currency: 'renminbi',
    currencyCode: 'CNY',
    rate: 3.278,
  },
  {
    country: 'Croatia',
    currency: 'kuna',
    currencyCode: 'HRK',
    rate: 3.235,
  },
];
