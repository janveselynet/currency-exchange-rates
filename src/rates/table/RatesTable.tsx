import React from 'react';
import styled from "styled-components";
import { Rate } from '../Rate';

const Table = styled.table`
  border: none;
  border-collapse:collapse;

  td,
  th {
    border: none;
    padding: 7px 10px;
  }

  thead > tr {
    font-weight: 900;
    color: #ffffff;
    background: #2980b9;
  }

  tbody tr {
    :nth-of-type(odd) {
      background: #e9e9e9;
    }

    td:last-child {
      text-align: right;
    }
  }
`;

interface RatesTableProps {
  rates: Array<Rate>;
};

export default function RatesTable({ rates }: RatesTableProps): JSX.Element {
  return (
    <Table>
      <thead>
        <tr>
          <th>Country</th>
          <th>Currency</th>
          <th>Currency code</th>
          <th>Rate</th>
        </tr>
      </thead>
      <tbody>
        {rates.map((rate, index) => (
          <tr key={index}>
            <td>{rate.country}</td>
            <td>{rate.currency}</td>
            <td>{rate.currencyCode}</td>
            <td>{rate.rate.toFixed(3)}</td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
}
