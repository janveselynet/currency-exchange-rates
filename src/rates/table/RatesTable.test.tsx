import { render } from '@testing-library/react';
import React from 'react';
import RatesTable from './RatesTable';
import { rates } from './__fixtures__/rates';

describe('RatesTable', () =>{
  it('renders table with currency exchange rates', () => {
    const { container } = render(<RatesTable rates={rates} />);
    expect(container).toMatchSnapshot();
  });
});
