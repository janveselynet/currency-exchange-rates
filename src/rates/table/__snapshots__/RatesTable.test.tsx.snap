// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatesTable renders table with currency exchange rates 1`] = `
<div>
  <table
    class="sc-bcXHqe iPxklV"
  >
    <thead>
      <tr>
        <th>
          Country
        </th>
        <th>
          Currency
        </th>
        <th>
          Amount
        </th>
        <th>
          Code
        </th>
        <th>
          Rate
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Australia
        </td>
        <td>
          dollar
        </td>
        <td>
          1
        </td>
        <td>
          AUD
        </td>
        <td>
          15.831
        </td>
      </tr>
      <tr>
        <td>
          Brazil
        </td>
        <td>
          real
        </td>
        <td>
          1
        </td>
        <td>
          BRL
        </td>
        <td>
          4.395
        </td>
      </tr>
      <tr>
        <td>
          Bulgaria
        </td>
        <td>
          lev
        </td>
        <td>
          1
        </td>
        <td>
          BGN
        </td>
        <td>
          12.473
        </td>
      </tr>
      <tr>
        <td>
          Canada
        </td>
        <td>
          dollar
        </td>
        <td>
          1
        </td>
        <td>
          CAD
        </td>
        <td>
          17.559
        </td>
      </tr>
      <tr>
        <td>
          China
        </td>
        <td>
          renminbi
        </td>
        <td>
          1
        </td>
        <td>
          CNY
        </td>
        <td>
          3.278
        </td>
      </tr>
      <tr>
        <td>
          Croatia
        </td>
        <td>
          kuna
        </td>
        <td>
          1
        </td>
        <td>
          HRK
        </td>
        <td>
          3.235
        </td>
      </tr>
      <tr>
        <td>
          Hungary
        </td>
        <td>
          forint
        </td>
        <td>
          100
        </td>
        <td>
          HUF
        </td>
        <td>
          5.923
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
